agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulsePingSender.java

```diff
@@ -19,68 +19,70 @@
+import com.azure.monitor.opentelemetry.exporter.implementation.quickpulse.model.QuickPulseEnvelope;
+import com.azure.monitor.opentelemetry.exporter.implementation.quickpulse.util.CustomCharacterEscapes;
+import com.azure.monitor.opentelemetry.exporter.implementation.utils.LazyHttpClient;
+import com.azure.monitor.opentelemetry.exporter.implementation.utils.Strings;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.SerializationFeature;
-import com.microsoft.applicationinsights.agent.internal.common.Strings;
-import com.microsoft.applicationinsights.agent.internal.httpclient.LazyHttpClient;
-import com.microsoft.applicationinsights.agent.internal.init.MainEntryPoint;
-import com.microsoft.applicationinsights.agent.internal.quickpulse.model.QuickPulseEnvelope;
-import com.microsoft.applicationinsights.agent.internal.quickpulse.util.CustomCharacterEscapes;
-import com.microsoft.applicationinsights.agent.internal.telemetry.TelemetryClient;
 import java.util.Date;
-import java.util.concurrent.atomic.AtomicBoolean;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 class QuickPulsePingSender {
 
   private static final Logger logger = LoggerFactory.getLogger(QuickPulsePingSender.class);
+  private static final String QP_BASE_URI =
+      "https://rt.services.visualstudio.com/QuickPulseService.svc";
+  private static final String LIVE_URL_PATH = "QuickPulseService.svc";
 
   private static final ObjectMapper mapper;
 
-  private static final String quickPulseVersion = MainEntryPoint.getAgentVersion();
+  // TODO Kishna populate this
+  private static final String quickPulseVersion = "(unknown)";
 
   private static final OperationLogger operationLogger =
       new OperationLogger(QuickPulsePingSender.class, "Pinging live metrics endpoint");
 
-  // TODO (kryalama) do we still need this AtomicBoolean, or can we use throttling built in to the
-  //  operationLogger?
-  private static final AtomicBoolean friendlyExceptionThrown = new AtomicBoolean();
-
   static {
     mapper = new ObjectMapper();
     mapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
     mapper.getFactory().setCharacterEscapes(new CustomCharacterEscapes());
   }
 
-  private final TelemetryClient telemetryClient;
   private final HttpPipeline httpPipeline;
   private final QuickPulseNetworkHelper networkHelper = new QuickPulseNetworkHelper();
   private volatile QuickPulseEnvelope pingEnvelope; // cached for performance
   private final String instanceName;
   private final String machineName;
   private final String quickPulseId;
+  private final String roleName;
+  private final String instrumentationKey;
+  private final String endPointUrl;
   private long lastValidTransmission = 0;
 
   public QuickPulsePingSender(
       HttpPipeline httpPipeline,
-      TelemetryClient telemetryClient,
+      String roleName,
+      String instrumentationKey,
```

Comment:
I think this needs `Supplier<String>` also since it can change at runtime in Azure Spring Cloud